### ISSUE TITLE
isChildOf() method always returns  `true`

### ISSUE THREAD

**Issue by texnixe** [getkirby/kirby#2255]
**Describe the bug**  
Whatever you pass to the `isChildMethod()` it always returns true.

Let's say you are in the Starterkit and add the following code in the `note.php`template:

```
var_dump($page->isChildOf('notes'));
```
rightly returns `true`, but

```
var_dump($page->isChildOf('photography'));
```
falsely also returns true. 

**Expected behavior**  
The method should create a page object from a string if a string is passed and then compare the parent to that the given page or the page created from the string.

**Kirby Version**  
3.0.0-rc-1

**Comment by afbora** [comment]
Isn't it natural that it always returns true? It was a self-validating method 😅 

    var_dump($page->isChildOf('no-exists-page')); // true :)

````php
public function isChildOf($parent): bool
{
    if ($parent = $this->parent()) {
        return $parent->is($parent);
    }

    return false;
}
````

https://github.com/getkirby/kirby/blob/release/3.3.0/src/Cms/Page.php#L632-L639

**Comment by texnixe** [comment]
You're right, I thought so. So the whole method is pretty useless as it is.

**Comment by lukasbestle** [comment]
OMG, that's a stupid bug. :D

**Comment by benoitwimart** [comment]
So, I really found a bug !

**Comment by bastianallgeier** [comment]
Sorry, but I really have to laugh about the stupidity here 😂 I think we can fix this for 3.3.0

**Comment by lukasbestle** [comment]
✅ Fixed on the 3.3.0 release branch.
